src/ragas/metrics/_answer_correctness.py

```diff
@@ -158,9 +159,9 @@ class AnswerCorrectness(MetricWithLLM, MetricWithEmbeddings):
 
     name: str = "answer_correctness"  # type: ignore[reportIncompatibleMethodOverride]
     evaluation_mode: EvaluationMode = EvaluationMode.qga  # type: ignore[reportIncompatibleMethodOverride]
-    correctness_prompt: Prompt = field(default_factory=lambda: CORRECTNESS_PROMPT)
+    correctness_prompt: Prompt = field(default_factory=correctness_prompt_factory)
     long_form_answer_prompt: Prompt = field(
-        default_factory=lambda: LONG_FORM_ANSWER_PROMPT
+        default_factory=long_form_answer_prompt_factory
```

Comment:
what if we make a factory inside `Prompt` object and call it `init()` or `factory()`.

Reply:
I think `init` goes a bit against the scope of `__init__`  and is misleading. The cleanest way would be to create a factory class, but that would add more complexity, so I'd say having a factory method is a good middle-way.